code/game/atoms.dm

```diff
@@ -152,6 +152,8 @@
 
 	/// list of clients that using this atom as their eye. SHOULD BE USED CAREFULLY
 	var/list/eye_users
+	/// same as 'eye_users', but mobs, instead of client. might be unstable.
+	var/list/eye_mobs
```

Comment:
These should be merged together. How is eye_mobs unstable?

Reply:
These are not mergeable. `eye_mobs` includes mobs that lost clients, and it's used to recover their eye back when its old client takes the mob back. So, the purpose of this variable is different than how `eye_users` works.
the reason why `eye_mobs` is unstable is because of harddel issue mainly. The logic of handling `eye_mobs` is alright, but when a code somewhere does a bad practice, the list will have some issue in regard to this.